File: backend/routes/app_state.py
from fastapi import APIRouter

from backend.models.app_state import get_app_state_store

router = APIRouter(prefix="/app-state", tags=["app-state"])


@router.get("")
async def get_app_state() -> dict:
    store = get_app_state_store()
    return store.get().to_dict()


@router.put("")
async def update_app_state(state: dict) -> dict:
    store = get_app_state_store()
    updated = store.update(state)
    return updated.to_dict()


@router.post("/recent-command")
async def add_recent_command(command: dict) -> dict:
    store = get_app_state_store()
    store.add_recent_command(command.get("id", ""), command.get("trigger", ""))
    return store.get().to_dict()

File: src/api/appState.ts
import { apiClient } from './client';

export interface AppState {
  last_voice_profile_id: string | null;
  last_tts_voice: string | null;
  window_position: { x: number | null; y: number | null };
  window_size: { width: number | null; height: number | null };
  active_command_ids: string[];
  recent_commands: Array<{
    id: string;
    trigger: string;
    used_at: string;
  }>;
  started_at: string | null;
  last_session_at: string | null;
}

export async function getAppState(): Promise<AppState> {
  return apiClient.get<AppState>('/app-state');
}

export async function updateAppState(state: Partial<AppState>): Promise<AppState> {
  return apiClient.put<AppState>('/app-state', state);
}

export async function addRecentCommand(command: { id: string; trigger: string }): Promise<AppState> {
  return apiClient.post<AppState>('/app-state/recent-command', command);
}
